project/CopyrightHeader.scala

```diff
@@ -40,55 +40,34 @@ trait CopyrightHeader extends AutoPlugin {
     s"Copyright (C) $year Lightbend Inc. <https://www.lightbend.com>"
 
-  val cStyleComment = HeaderCommentStyle.cStyleBlockComment.copy(commentCreator = new CommentCreator() {
-
-    def updateLightbendHeader(header: String): String = header match {
-      case CopyrightHeaderPattern(years, null, _) =>
-        if (years != CurrentYear)
-          CopyrightPattern.replaceFirstIn(header, headerFor(years + "-" + CurrentYear))
-        else
-          CopyrightPattern.replaceFirstIn(header, headerFor(years))
-      case CopyrightHeaderPattern(years, endYears, _) =>
-        CopyrightPattern.replaceFirstIn(header, headerFor(years.replace(endYears, "-" + CurrentYear)))
-      case _ =>
-        header
-    }
+  def apacheHeader: String =
+    """Licensed to the Apache Software Foundation (ASF) under one
+      |or more contributor license agreements.  See the NOTICE file
+      |distributed with this work for additional information
+      |regarding copyright ownership.  The ASF licenses this file
+      |to you under the Apache License, Version 2.0 (the
+      |"License"); you may not use this file except in compliance
+      |with the License.  You may obtain a copy of the License at
+      |
+      |  http://www.apache.org/licenses/LICENSE-2.0
+      |
+      |Unless required by applicable law or agreed to in writing,
+      |software distributed under the License is distributed on an
+      |"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+      |KIND, either express or implied.  See the License for the
+      |specific language governing permissions and limitations
+      |under the License.
+      |""".stripMargin
 
-    def parseStartAndEndYear(header: String): Option[(String, Option[String])] = header match {
-      case CopyrightHeaderPattern(years, null, _) =>
-        Some((years, None))
-      case CopyrightHeaderPattern(years, endYears, _) =>
-        Some((years, Some(endYears)))
-      case _ =>
-        None
-    }
+  val cStyleComment = HeaderCommentStyle.cStyleBlockComment.copy(commentCreator = new CommentCreator() {
 
     override def apply(text: String, existingText: Option[String]): String = {
       val formatted = existingText match {
         case Some(existedText) =>
-          parseStartAndEndYear(existedText) match {
-            case Some((years, None)) =>
-              if (years != CurrentYear) {
-                val header = headerFor(years + "-" + CurrentYear)
-                HeaderCommentStyle.cStyleBlockComment.commentCreator(header, existingText)
-              } else {
-                HeaderCommentStyle.cStyleBlockComment.commentCreator(headerFor(CurrentYear), existingText)
-              }
-            case Some((years, Some(endYears))) =>
-              val header = headerFor(years.replace(endYears, "-" + CurrentYear))
-              HeaderCommentStyle.cStyleBlockComment.commentCreator(header, existingText)
-            case None =>
-              existedText
-          }
+          existedText
```

Comment:
Is it possible to be a lot more clear here about whats going on with the headers considering we have a non trivial setup regarding different types of headers, i.e. something like

```scala
override def apply(text: String, existingText: Option[String]): String = {
  val formatted = existingText match {
    case Some(existingText) if existedText.contains("Licensed to the Apache Software Foundation (ASF)") && existingText.contains("Copyright") =>
      // This is the case where we have a major change in a previously existing source file with a copyright header, usually (but not always) Lightbend
      existingText
    case Some(existedText) if existingText.contains("Copyright")=>
      existedText
    case Some(existingText) if existedText.contains("Licensed to the Apache Software Foundation (ASF)") =>
      existingText
    case Some(_) =>
      throw new Exception(s"Unknown header detected: $existedText")
    case None =>
      HeaderCommentStyle.cStyleBlockComment.commentCreator(text, existingText)
}
```

Reply:
I see, I will try to update this after work.

Reply:
> Is contains checking case sensitive? If not the Apache license contains the word "copyright" and may cause a problem.

No its not but that can easily be remedied, i.e.

```scala
case Some(existingText) if existedText.toLowerCase.contains("Licensed to the Apache Software Foundation (ASF)") && existingText.contains("Copyright")
```

Reply:
Do we need to update the lightbend's year?

Reply:
the Lightbend header should not be editted at all - whatever year is in the file now should stay that way indefinitely